### ISSUE TITLE
Can't create directive with a Scalar that has an applied directive

### ISSUE THREAD

**Issue by etienne-sf** [graphql-java/graphql-java#2055]
**Describe the bug**
When a Scalar definition has a directive applied, this Scalar can not be used as a parameter in another directive.
This generates a "java.util.NoSuchElementException: No value present" exception when building the schema

**To Reproduce**
Using the GraphQL schema allows to repeat this bug:

```
scalar Date @DummyDirective  #Everything works nice if this @DummyDirective is not applied here

# A test directive, that can be added anywhere
directive @testDirective(aDate: Date) on OBJECT

directive @DummyDirective on SCALAR 

type Query {
	aQuery: String   
}
```

**Comment by etienne-sf** [comment]
This was tested with the graphql-java 15.0 release

**Comment by etienne-sf** [comment]
This can be repeated on enum definitions:

```
enum Episode @DummyDirective #Everything works nice if this @DummyDirective is not applied
{
	JEDI
	NEWHOPE
}

directive @testDirective(anEnum: Episode = JEDI) on OBJECT

directive @DummyDirective on ENUM


type Query {
	aQuery: String   
}
```

**Comment by etienne-sf** [comment]
Same issue with enum values:


```
directive @testDirective(anEnum: Episode = JEDI) on OBJECT

directive @DummyDirective on ENUM_VALUE

enum Episode   {
    NEWHOPE
    EMPIRE   @DummyDirective
    JEDI
    DOES_NOT_EXIST  
}

type Query {
	aQuery: String   
}
```

**Comment by dugenkui03** [comment]
@bbakerman 
The reason of this bug is that: the argument of directive @testDirective is customized scalar Date, and buildDirective(x) can not find directive @DummyDirective from empty list when create GraphQLDirective used by argument of customized scalar Date.

I submitted a PR for this, please help me determine whether it is a appropriate way to fix this.

**Comment by etienne-sf** [comment]
Sorry for the delay.

Yes, this build solved my issue.
All my test cases on this one are solved, when switching from graphql-java 15.0 to this local build.

There is no emergency on my side, but do you know when an official package will be available with this correction ?

Thank you.
Etienne





BTW, for info (and for my own memory) :

I had some issues, when compiling your PR.

It won't compile on m computer, with JDK 13.
And with jdk 8, I got some test errors.

I check out the _bugFix_for_2055_ branch from _https://github.com/dugenkui03/graphql-java.git_
(for info, and for my own memory) Then I executed:
gradlew build  -x test
Then:
gradlew publishToMavenLocal -x test
(as I use it in a maven project)

I guess the last line is enough.

**Comment by dugenkui03** [comment]
The official version will released in 2020.12 or 2021.01, I guess.

It seem that each of the commit which merged in master will updated in maven repository.
